### PR TITLE
docs(requirements): update optimize requirement with Java 21

### PR DESCRIPTION
**Pull request by buccarel**
## Description

Optimize is dropping the support for Java 17 and 20, in favor of Java 21.
With this PR, the documentation will be updated accordingly.
Issue: https://github.com/camunda/camunda-optimize/issues/12541

## When should this change go live?
Contextually with the release of Optimize 3.14 (scheduled for 8th of October 2024)

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineerring DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer.

### CODE REVIEW

File: versioned_docs/version-8.5/reference/supported-environments.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ Requirements for the components can be seen below:
...
 | Tasklist    | OpenJDK 17+  | Elasticsearch 8.9+<br/>Amazon OpenSearch 2.5.x                                                                                                                                                                    |
 | Identity    | OpenJDK 17+  | Keycloak 22.x, 23.x<br/>PostgreSQL 14.x, 15.x or Amazon Aurora PostgreSQL 13.x, 14.x, 15.x (required for [certain features](/self-managed/identity/deployment/configuration-variables.md#database-configuration)) |
-| Optimize    | OpenJDK 17+  | Elasticsearch 8.9+<br/>Amazon OpenSearch 2.5.x                                                                                                                                                                    |
+| Optimize    | OpenJDK 21+  | Elasticsearch 8.9+<br/>Amazon OpenSearch 2.5.x                                                                                                                                                                    |
```
Comment (RomanJRW) [inline versioned_docs/version-8.5/reference/supported-environments.md:93]: these are versioned docs for 8.5. Given that we don't want to backport this, let's not change this file

**Review by RomanJRW — CHANGES_REQUESTED** [review]
Thanks for doing the docs for this too!

**Comment by christinaausley** [comment]
Hi 👋 

Is this a breaking change? Do we need to note this elsewhere in the docs, or in the announcements?

**Comment by RomanJRW** [comment]
@christinaausley - is it enough that we announced it [here](https://github.com/camunda/camunda-docs-static/pull/463)? Or do you mean a different announcements page?

**Pull request merged**